src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -252,29 +252,18 @@ end
 
 function check_periodicity_mesh_boundary_conditions(mesh::Union{TreeMesh{1},
                                                                 StructuredMesh{1}},
-                                                    boundary_conditions::NamedTuple)
-    @unpack x_neg, x_pos = boundary_conditions
-    check_periodicity_mesh_boundary_conditions_x(mesh, x_neg, x_pos)
-end
-
-function check_periodicity_mesh_boundary_conditions(mesh::Union{TreeMesh{1},
-                                                                StructuredMesh{1}},
-                                                    boundary_conditions::NTuple{2, Any})
+                                                    boundary_conditions::Union{NamedTuple,
+                                                                               NTuple{2,
+                                                                                      Any}})
```

Comment:
Do you need the restriction on `boundary_conditions` here or could you also use

```suggestion
                                                    boundary_conditions)
```

?

Reply:
In that case I get an error:

```julia
  LoadError: MethodError: check_periodicity_mesh_boundary_conditions(::Trixi.StructuredMesh{1, Float64}, ::Trixi.BoundaryConditionPeriodic) is ambiguous.
  
  Candidates:
    check_periodicity_mesh_boundary_conditions(mesh::Union{Trixi.StructuredMesh, Trixi.TreeMesh}, boundary_conditions::Trixi.BoundaryConditionPeriodic)
      @ Trixi ~/git/Trixi.jl/src/semidiscretization/semidiscretization_hyperbolic.jl:224
    check_periodicity_mesh_boundary_conditions(mesh::Union{Trixi.StructuredMesh{1}, Trixi.TreeMesh{1, TreeType} where TreeType<:Trixi.AbstractTree{1}}, boundary_conditions)
      @ Trixi ~/git/Trixi.jl/src/semidiscretization/semidiscretization_hyperbolic.jl:253
      ```
      
      Looks like the "specialization" based on the mesh dimensionality is not enough

Reply:
What if you list the 1D cases explicitly in the specialization for `boundary_conditions::BoundaryConditionPeriodic`? Does that resolve the ambiguity?

Reply:
Hm, I guess that should work - but this leads to more versions of the function for `BoundaryConditionPeriodic` for each dimension. 
Is that preferable over the dispatch on `boundary_conditions`?

Reply:
Ok, you're right. But can you maybe reduce the specialization to
```suggestion
                                                    boundary_conditions::Union{Tuple, NamedTuple})
```

?

Reply:
That seems to work :+1:

Reply:
Thanks!